src/definition/accessors/IUserRead.ts

```diff
@@ -8,4 +8,6 @@ export interface IUserRead {
     getById(id: string): Promise<IUser>;
 
     getByUsername(username: string): Promise<IUser>;
+
+    createBot(username: string, email: string, name: string, password: string): Promise<void>;
```

Comment:
I think we would have to create another interface, maybe a `IUserCreator`, to define this method.

Also, you could actually receive as a parameter either a full `IUser` object or a subset of that, for instance `IBotUser`, containing only the properties that you need.